src/primitives/block.cpp

```diff
@@ -27,20 +29,21 @@ uint256 CBlockHeaderUncached::GetHash() const
 // yespower
 uint256 CBlockHeaderUncached::GetPoWHash() const
 {
-    uint256 thash;
-    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
-    ss << *this;
-    yespower_params_t yespower_1_0_sugarchain = {
+    static const yespower_params_t yespower_1_0_sugarchain = {
         .version = YESPOWER_1_0,
         .N = 2048,
         .r = 32,
         .pers = (const uint8_t *)"Satoshi Nakamoto 31/Oct/2008 Proof-of-work is essentially one-CPU-one-vote",
         .perslen = 74
     };
-    if (yespower_tls( (unsigned char *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&thash) ) {
-        abort();
+    uint256 hash;
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    ss << *this;
+    if (yespower_tls((unsigned char *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&hash)) {
```

Comment:
@solardiz should we use `(const uint8_t *)` instead of `(unsigned char *)`?

Reply:
https://github.com/sugarchain-project/sugarchain/blob/0624be48d82c38367c108b84c5532ca95e30263f/src/crypto/yespower-1.0.1/yespower-opt.c#L1129-L1141